mercury-operator/src/main/java/com/redhat/mercury/operator/event/DeploymentEvent.java

```diff
@@ -0,0 +1,30 @@
+package com.redhat.mercury.operator.event;
+
+import io.fabric8.kubernetes.api.model.apps.Deployment;
+import io.fabric8.kubernetes.client.Watcher;
+import io.javaoperatorsdk.operator.processing.event.DefaultEvent;
+import io.javaoperatorsdk.operator.processing.event.EventSource;
+
+public class DeploymentEvent extends DefaultEvent {
+
+    private final Watcher.Action action;
+    private final Deployment deployment;
+
+    public DeploymentEvent(Watcher.Action action, Deployment resource, EventSource eventSource) {
+        super(resource.getMetadata().getOwnerReferences().get(0).getUid(), eventSource);
```

Comment:
I have seen resources with many ownerReferences and that might cause conflicts or bugs, change the `get(0).getUid()` and get the OwnerReference that is of type `ServiceDomain`
Check if you do the same in other parts of the code.